internal/service/remotecfg/remotecfg.go

```diff
@@ -166,6 +167,17 @@ func New(opts Options) (*Service, error) {
 		opts:        opts,
 		systemAttrs: getSystemAttributes(),
 		ticker:      jitter.NewTicker(math.MaxInt64-baseJitter, baseJitter), // first argument is set as-is to avoid overflowing
+		clientFactory: func(args Arguments) (collectorv1connect.CollectorServiceClient, error) {
```

Comment:
This refactor was required to make tests pass, otherwise the initial testEnv update call got a 403.